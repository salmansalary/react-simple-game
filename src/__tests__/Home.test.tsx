import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Home from '../routes/home';

describe('Home', () => {

    it('should render HOME correctly', () => {
        const component = shallow(<Home />);
    
        expect(component).toMatchSnapshot();
    });

    it('should stop navigating to play if NAME input is empty and show warning', () => {
        const component = mount(<Home />);
        component
        .find('button.btnStart')
        .simulate('click');

        expect(component.find('input.warning').length).toBe(1);
    });

});
