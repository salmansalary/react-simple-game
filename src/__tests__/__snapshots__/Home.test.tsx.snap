// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render HOME correctly 1`] = `ShallowWrapper {}`;
